In [1]:
import pandas as pd
import re

In [2]:
netflix_data = pd.read_csv('../data/raw/netflix_branded_titles_with_release_year.csv')

In [4]:
#split combined_title into title and season
netflix_data.rename(columns={'title_desc':'combined_title'}, inplace=True)
netflix_data[['title', 'season']] = netflix_data['combined_title'].str.split(': Season ', expand=True)
netflix_data['title'] = netflix_data['title'].str.lower()
netflix_data['is_tv_show'] = netflix_data['season'].notnull()
netflix_data['release_year'] = netflix_data['release_year'].astype('int')

In [134]:
#regex to get title or season from combined_title
netflix_data['title'] = netflix_data['combined_title'].str.extract(r'(.+?)($|: Season \d+)', expand=False)[0]
netflix_data['title'] = netflix_data['title'].str.lower()
netflix_data['season'] = netflix_data['combined_title'].str.extract(r': Season (\d+)')

In [2]:
def clean_title(title):
    # Remove punctuation
    title = title.str.replace(r'[^\w\s]', '', regex=True)
    # Remove leading and trailing whitespace
    title = title.str.strip()
    # Remove leading and trailing whitespace
    title = title.str.lower()
    return title

In [145]:
#apply clean title
netflix_data['title'] = clean_title(netflix_data['title'])

In [146]:
netflix_data[netflix_data['title'].str.contains('journeys')]

combined_title  release_year  release_date  \
1336         Pokémon Journeys: The Series: Part 2          2020      20200911   
1337  Pokémon Master Journeys: The Series: Part 1          2021      20210910   
1338  Pokémon Master Journeys: The Series: Part 3          2022      20220526   

      is_top10                                      title season  is_tv_show  
1336      True         pokémon journeys the series part 2    NaN       False  
1337      True  pokémon master journeys the series part 1    NaN       False  
1338      True  pokémon master journeys the series part 3    NaN       False

In [2]:
yendata = pd.read_csv("../data/interim/movies_matched.csv")

In [124]:
manual_tv_shows = {'pokémon journeys: the series: part 2': 2,
    'pokémon master journeys: the series: part 1': 1,
    'pokémon master journeys: the series: part 3': 3,
    'pretend it’s a city: limited series': 1,
    "rilakkuma's theme park adventure: シーズン１": 1,
    '\u200bsaint seiya: knights of the zodiac: part ii': 2,
    'tiger & bunny 2': 2}

In [140]:
netflix_data[netflix_data['title'].str.contains('pretend it')]

combined_title  release_year  release_date  \
1348  Pretend It’s a City: Limited Series          2021      20210108   

      is_top10                                title season  is_tv_show  
1348     False  pretend it’s a city: limited series    NaN       False

In [3]:
netflix_data = pd.read_csv("../data/interim/netflix_branded_titles_test.csv")

In [4]:
manual = pd.read_csv("../data/interim/multimatch_manual.csv")

In [5]:
movies_and_season_1 = netflix_data[(netflix_data['is_tv_show'] == False) | (netflix_data['season'] == 1)]
tv_shows_and_high_seasons = netflix_data[(netflix_data['is_tv_show'] == True) & (netflix_data['season'] > 1)]

In [6]:
movies_only = netflix_data[(netflix_data['is_tv_show'] == False)]
tv_shows_only = netflix_data[(netflix_data['is_tv_show'] == True)]

In [16]:
imdb_all = pd.read_csv("../data/processed/imdb_all.csv").rename(columns={'startYear':'release_year'})
imdb_all['title'] = clean_title(imdb_all['title'])

In [20]:
movie_merge = pd.merge(movies_only, imdb_all[~imdb_all['titleType'].isin(['tvSeries', 'video', 'videoGame', 'tvPilot', 'tvMiniSeries'])], on=['title', 'release_year'], how='left', indicator=True)

In [17]:
import numpy as np

In [21]:
movie_good = movie_merge[movie_merge['_merge'] == 'both'].groupby('combined_title').filter(lambda x: len(x) == 1)
movie_multi = movie_merge[movie_merge['_merge'] == 'both'].groupby('combined_title').filter(lambda x: len(x) > 1)['combined_title'].unique()
movie_unmatched = movie_merge[movie_merge['_merge'] == 'left_only']['combined_title'].unique()
manual[manual['combined_title'].isin(np.append(movie_multi, movie_unmatched))].to_clipboard(index=False)

In [22]:
tv_show_merge = pd.merge(tv_shows_only, imdb_all[imdb_all['titleType'].isin(['tvSeries', 'tvMiniSeries'])], on=['title'], how='left', indicator=True)

In [23]:
tv_show_good = tv_show_merge[tv_show_merge['_merge'] == 'both'].groupby('combined_title').filter(lambda x: len(x) == 1)
tv_show_multi = tv_show_merge[tv_show_merge['_merge'] == 'both'].groupby('combined_title').filter(lambda x: len(x) > 1)['combined_title'].unique()
tv_show_unmatched = tv_show_merge[tv_show_merge['_merge'] == 'left_only']['combined_title'].unique()
manual[manual['combined_title'].isin(np.append(tv_show_multi, tv_show_unmatched))].to_clipboard(index=False)

In [24]:
automated_tconsts = pd.concat([movie_good[['combined_title', 'tconst']], tv_show_good[['combined_title', 'tconst']]], axis=0)
manual_tconsts = pd.read_csv('../data/interim/manual.csv')

tconst_joiner = pd.concat([automated_tconsts, manual_tconsts], axis=0)

In [35]:
pd.merge(netflix_data, tconst_joiner, on=['combined_title'], how='left', indicator=True).to_csv('../data/final_data/joseph_check.csv', index=False)

In [36]:
imdb_all[imdb_all['tconst']=='tt13666778']

Empty DataFrame
Columns: [tconst, titleType, primaryTitle, originalTitle, isAdult, release_year, endYear, runtimeMinutes, genres, title]
Index: []

In [14]:
imdb_all[imdb_all['tconst']=='tt19187166']

Empty DataFrame
Columns: [tconst, titleType, primaryTitle, originalTitle, isAdult, startYear, endYear, runtimeMinutes, genres]
Index: []

In [13]:
movies_and_season_1_merge = pd.merge(movies_and_season_1, imdb_all, on=['release_year', 'title'], how='left', indicator=True)

In [59]:
#1256 everything is good
good_merges_1 = movies_and_season_1_merge[movies_and_season_1_merge['_merge'] == 'both'].groupby('combined_title').filter(lambda x: len(x) == 1)
#good_merges.to_csv("../data/interim/1256_good_matches.csv")
#good_merges_1[(~good_merges_1['is_tv_show'])]['titleType'].value_counts()

In [15]:
movies_and_season_1_merge[movies_and_season_1_merge['_merge'] == 'both'].groupby('combined_title').filter(lambda x: len(x) > 1)['combined_title'].unique()


array(['Alba: Season 1', 'All Together Now', 'Alone', 'Animal: Season 1',
       'As the Crow Flies: Season 1', 'Ava', 'Away: Season 1',
       'Baby Fever: Season 1', 'Bad Trip', 'Beats', 'Becoming',
       'Brave New World: Season 1', 'Break Point: Season 1', 'Bubble',
       'Canvas', 'Carter', 'Catching Killers: Season 1', 'Cobalt Blue',
       'Connected: Season 1', 'Cops and Robbers', 'Crush',
       'Cursed: Season 1', 'Cuties', 'DNA', 'Dangerous Liaisons',
       'Dark Desire: Season 1', 'Darlings', 'Dealer: Season 1', 'Deep',
       'Disclosure', 'Dracula: Season 1', 'Eden: Season 1',
       'Equinox: Season 1', 'Ethos: Season 1',
       'Everything Will Be Fine: Season 1', 'Extraction', 'Fatherhood',
       'Feel Good: Season 1', 'Fever Dream', 'Fierce',
       'Followers: Season 1', 'Forgive Us Our Trespasses', 'Found',
       'Godspeed', 'Guilty', 'HOMUNCULUS', 'Heist: Season 1',
       'Hollywood: Season 1', 'Horse Girl', 'Incantation',
       'Intimacy: Season 1', 'Into t

In [60]:
movies_and_season_1_merge[movies_and_season_1_merge['_merge'] == 'both'].groupby('combined_title').filter(lambda x: len(x) > 1)['combined_title'].isin(manual['combined_title']).all()

True

In [61]:
bad_merges_1 = movies_and_season_1_merge[movies_and_season_1_merge['_merge'] != 'both']



In [63]:
bad_merges_1['combined_title'].isin(manual['combined_title']).all()

True

In [66]:
tv_shows_and_high_seasons_merge = pd.merge(tv_shows_and_high_seasons, imdb_all, on=['title'], how='left', indicator=True)

In [70]:
good2 = tv_shows_and_high_seasons[tv_shows_and_high_seasons['combined_title'].isin(manual['combined_title'])]

In [73]:
high_seasons_merge = tv_shows_and_high_seasons[~tv_shows_and_high_seasons['combined_title'].isin(manual['combined_title'])].merge(imdb_all, on=['title'], how='left', indicator=True)

In [75]:
only1match_high_seasons = high_seasons_merge.groupby('combined_title').filter(lambda x: len(x) == 1)

In [81]:
multiple_matches_high_seasons = high_seasons_merge.groupby('combined_title').filter(lambda x: len(x) > 1)

multiple_matches_high_seasons[multiple_matches_high_seasons['title'].isin(good_merges_1['title'])]

combined_title  release_year_x  \
41                              Alien TV: Season 2            2021   
42                              Alien TV: Season 2            2021   
102                        Blood & Water: Season 2            2021   
103                        Blood & Water: Season 2            2021   
159                            Control Z: Season 2            2021   
160                            Control Z: Season 2            2021   
161                            Control Z: Season 3            2022   
162                            Control Z: Season 3            2022   
255                          Locke & Key: Season 2            2021   
256                          Locke & Key: Season 2            2021   
257                          Locke & Key: Season 3            2022   
258                          Locke & Key: Season 3            2022   
346                             Ragnarok: Season 2            2021   
347                             Ragnarok: Season 2            2021   
348                             Ragnarok: Season 2            2021   
349                             Ragnarok: Season 2            2021   
362  RuPaul's Secret Celebrity Drag Race: Season 2            2022   
363  RuPaul's Secret Celebrity Drag Race: Season 2            2022   
381                          Space Force: Season 2            2022   
382                          Space Force: Season 2            2022   
383                          Space Force: Season 2            2022   
384                          Space Force: Season 2            2022   
385                             Spycraft: Season 2            2022   
386                             Spycraft: Season 2            2022   

     release_date  is_top10                               title  season  \
41       20210319     False                            alien tv     2.0   
42       20210319     False                            alien tv     2.0   
102      20210924      True                        blood  water     2.0   
103      20210924      True                        blood  water     2.0   
159      20210804      True                           control z     2.0   
160      20210804      True                           control z     2.0   
161      20220706      True                           control z     3.0   
162      20220706      True                           control z     3.0   
255      20211022      True                          locke  key     2.0   
256      20211022      True                          locke  key     2.0   
257      20220810      True                          locke  key     3.0   
258      20220810      True                          locke  key     3.0   
346      20210527      True                            ragnarok     2.0   
347      20210527      True                            ragnarok     2.0   
348      20210527      True                            ragnarok     2.0   
349      20210527      True                            ragnarok     2.0   
362      20220812     False  rupauls secret celebrity drag race     2.0   
363      20220812     False  rupauls secret celebrity drag race     2.0   
381      20220218      True                         space force     2.0   
382      20220218      True                         space force     2.0   
383      20220218      True                         space force     2.0   
384      20220218      True                         space force     2.0   
385      20220111      True                            spycraft     2.0   
386      20220111      True                            spycraft     2.0   

     is_tv_show      tconst titleType                         primaryTitle  \
41         True  tt10910330     short                             Alien TV   
42         True  tt12765530  tvSeries                             Alien TV   
102        True  tt14810192     short                        Blood & Water   
103        True   tt9839146  tvSeries                        Blood & Water   
159        True  tt

In [119]:
good_merges_1[good_merges_1['title'].str.contains("rupaul")]

combined_title  release_year  \
1334  RuPaul's Secret Celebrity Drag Race: Season 1          2020   

      release_date  is_top10                               title  season  \
1334      20200424     False  rupauls secret celebrity drag race     1.0   

      is_tv_show      tconst titleType                         primaryTitle  \
1334        True  tt11187480  tvSeries  RuPaul's Secret Celebrity Drag Race   

                            originalTitle  isAdult endYear runtimeMinutes  \
1334  RuPaul's Secret Celebrity Drag Race      0.0      \N             63   

                    genres _merge  
1334  Game-Show,Reality-TV   both

In [112]:
TV_only_1 = multiple_matches_high_seasons[multiple_matches_high_seasons['titleType'].isin(['tvSeries', 'miniTVSeries'])].groupby('combined_title').filter(lambda x: len(x) == 1)

In [117]:
multiple_matches_high_seasons[multiple_matches_high_seasons['titleType'].isin(['tvSeries', 'miniTVSeries'])].groupby('combined_title').filter(lambda x: len(x) > 1)

combined_title  release_year_x  \
133                     Can You Hear Me?: Season 2            2020   
142                     Can You Hear Me?: Season 2            2020   
362  RuPaul's Secret Celebrity Drag Race: Season 2            2022   
363  RuPaul's Secret Celebrity Drag Race: Season 2            2022   
415                         The Neighbor: Season 2            2021   
423                         The Neighbor: Season 2            2021   

     release_date  is_top10                               title  season  \
133      20200113     False                     can you hear me     2.0   
142      20200113     False                     can you hear me     2.0   
362      20220812     False  rupauls secret celebrity drag race     2.0   
363      20220812     False  rupauls secret celebrity drag race     2.0   
415      20210521     False                        the neighbor     2.0   
423      20210521     False                        the neighbor     2.0   

     is_tv_show      tconst titleType                         primaryTitle  \
133        True  tt15228538  tvSeries                     Can You Hear Me?   
142        True   tt9747016  tvSeries                      Can You Hear Me   
362        True  tt11187480  tvSeries  RuPaul's Secret Celebrity Drag Race   
363        True  tt21342754  tvSeries  RuPaul's Secret Celebrity Drag Race   
415        True  tt21810202  tvSeries                         The Neighbor   
423        True   tt9731222  tvSeries                         The Neighbor   

                           originalTitle  isAdult  release_year_y endYear  \
133                     Can You Hear Me?      0.0          2020.0      \N   
142                        M'entends-tu?      0.0          2018.0    2021   
362  RuPaul's Secret Celebrity Drag Race      0.0          2020.0      \N   
363  RuPaul's Secret Celebrity Drag Race      0.0          2022.0      \N   
415                            Ham Sayeh      0.0          2021.0    2021   
423                            El vecino      0.0          2019.0    2021   

    runtimeMinutes                genres _merge  
133             20        Drama,Thriller   both  
142             30          Comedy,Drama   both  
362             63  Game-Show,Reality-TV   both  
363             \N                Family   both  
415             \N                 Drama   both  
423             30         Action,Comedy   both

In [28]:
dupes1[~dupes1['combined_title']

Empty DataFrame
Columns: [combined_title, release_year, release_date, is_top10, title, season, is_tv_show, tconst, titleType, primaryTitle, originalTitle, isAdult, endYear, runtimeMinutes, genres, _merge]
Index: []

In [98]:
movies_matched = pd.read_csv("../data/interim/movies_matched.csv")
tv_matched = pd.read_csv("../data/interim/tv_matched.csv")
final_match = pd.read_csv("../data/final_data/netflix_final_data.csv")

In [99]:
pd.merge(movies_matched, final_match, on=['title'])[['combined_title', 'tconst_y']].to_csv("../data/interim/multimatch_manual2.csv", index=False)
pd.merge(tv_matched, final_match, on=['title'])[['combined_title', 'tconst_y']].to_csv("../data/interim/multimatch_manual3.csv", index=False)

In [101]:
good_merges_1.titleType.value_counts()

tvSeries        519
movie           512
tvSpecial       116
tvMiniSeries     50
short            39
tvMovie          17
tvShort           2
video             1
Name: titleType, dtype: int64

In [110]:
good_merges_1[(good_merges_1['titleType']=='tvMiniSeries') & (good_merges_1['is_tv_show'] == False)]

combined_title  release_year  release_date  is_top10  \
617                           Furioza          2022      20220406      True   
1018              Measure for Measure          2021      20210101     False   
1668        The Haunting of Bly Manor          2020      20201009      True   
1876  Tiger King: The Doc Antle Story          2021      20211210      True   
2079       jeen-yuhs: A Kanye Trilogy          2022      20220216      True   

                               title  season  is_tv_show      tconst  \
617                          furioza     NaN       False  tt18286666   
1018             measure for measure     NaN       False  tt22456470   
1668       the haunting of bly manor     NaN       False  tt10970552   
1876  tiger king the doc antle story     NaN       False  tt16307970   
2079        jeenyuhs a kanye trilogy     NaN       False  tt14599438   

         titleType                     primaryTitle  \
617   tvMiniSeries                          Furioza   
1018  tvMiniSeries              Measure for Measure   
1668  tvMiniSeries        The Haunting of Bly Manor   
1876  tvMiniSeries  Tiger King: The Doc Antle Story   
2079  tvMiniSeries       Jeen-yuhs: A Kanye Trilogy   

                        originalTitle  isAdult endYear runtimeMinutes  \
617                           Furioza      0.0    2022             40   
1018              Measure for Measure      0.0    2021             \N   
1668        The Haunting of Bly Manor      0.0    2020            494   
1876  Tiger King: The Doc Antle Story      0.0    2021            127   
2079       Jeen-yuhs: A Kanye Trilogy      0.0    2022             90   

                           genres _merge  
617                         Crime   both  
1018                        Drama   both  
1668         Drama,Horror,Mystery   both  
1876  Biography,Crime,Documentary   both  
2079            Documentary,Music   both

In [38]:
multimatch1 = movies_and_season_1_merge[movies_and_season_1_merge['_merge'] == 'both'].groupby('combined_title').filter(lambda x: len(x) > 1)

In [52]:
movies_and_season_1[movies_and_season_1['combined_title'].str.contains("The One:")]

combined_title  release_year  release_date  is_top10    title  \
1879  The One: Season 1          2021      20210616      True  the one   
1880  The One: Season 1          2021      20210312      True  the one   

      season  is_tv_show  
1879     1.0        True  
1880     1.0        True

In [42]:
pd.DataFrame({"combined_title":multimatch1['combined_title'].unique()}).to_csv("../data/interim/multimatch_manual.csv")

In [53]:
nomatch = movies_and_season_1_merge[movies_and_season_1_merge['_merge'] == 'left_only']

In [54]:
nomatch

combined_title  release_year  \
2                         10 Days With Santa Claus          2020   
3                             100 Humans: Season 1          2020   
11                                      37 Seconds          2020   
16                                  7SEEDS: Part 2          2020   
17                                               8          2020   
...                                            ...           ...   
2066              Young Wallander: Killer's Shadow          2022   
2086   the goop lab with Gwyneth Paltrow: Season 1          2020   
2087                            tick, tick...BOOM!          2021   
2088                                       Òlòtūré          2020   
2089  ​SAINT SEIYA: Knights of the Zodiac: Part II          2020   

      release_date  is_top10                                      title  \
2         20201204     False                   10 days with santa claus   
3         20200313     False                                 100 humans   
11        20200131     False                                 37 seconds   
16        20200326      True                              7seeds part 2   
17        20200612     False                                          8   
...            ...       ...                                        ...   
2066      20220217      True             young wallander killers shadow   
2086      20200124     False          the goop lab with gwyneth paltrow   
2087      20211119      True                              tick tickboom   
2088      20201002     False                                    òlòtūré   
2089      20200123      True  saint seiya knights of the zodiac part ii   

      season  is_tv_show tconst titleType primaryTitle originalTitle  isAdult  \
2        NaN       False    NaN       NaN          NaN           NaN      NaN   
3        1.0        True    NaN       NaN          NaN           NaN      NaN   
11       NaN       False    NaN       NaN          NaN           NaN      NaN   
16       NaN       False    NaN       NaN          NaN           NaN      NaN   
17       NaN       False    NaN       NaN          NaN           NaN      NaN   
...      ...         ...    ...       ...          ...           ...      ...   
2066     NaN       False    NaN       NaN          NaN           NaN      NaN   
2086     1.0        True    NaN       NaN          NaN           NaN      NaN   
2087     NaN       False    NaN       NaN          NaN           NaN      NaN   
2088     NaN       False    NaN       NaN          NaN           NaN      NaN   
2089     NaN       False    NaN       NaN          NaN           NaN      NaN   

     endYear runtimeMinutes genres     _merge  
2        NaN            NaN    NaN  left_only  
3        NaN            NaN    NaN  left_only  
11       NaN            NaN    NaN  left_only  
16       NaN            NaN    NaN  left_only  
17       NaN            NaN    NaN  left_only  
...      ...            ...    ...        ...  
2066     NaN            NaN    NaN  left_only  
2086     NaN            NaN    NaN  left_only  
2087     NaN            NaN    NaN  left_only  
2088     NaN            NaN    NaN  left_only  
2089     NaN            NaN    NaN  left_only  

[414 rows x 16 columns]

In [40]:
multimatch1.to_csv("../data/interim/multimatch1.csv")

In [37]:
good_merges_1[good_merges_1['titleType']=='short']

combined_title  release_year  \
46                          A StoryBots Space Adventure          2021   
170                                             Audible          2021   
218                 Behind the Scenes With Jane Campion          2022   
246                                Blood Ties: Season 1          2022   
274                                             Bruised          2021   
294           Camp Confidential: America's Secret Nazis          2021   
302               Captain Underpants Epic Choice-o-Rama          2020   
306           Carmen Sandiego: To Steal or Not to Steal          2020   
311                                         Cat Burglar          2022   
322             Chadwick Boseman: Portrait of an Artist          2021   
529                                                Erax          2022   
532                               Escape The Undertaker          2021   
647                    Go! Go! Cory Carson: The Chrissy          2020   
689                         Headspace: Unwind Your Mind          2021   
691                                          Heart Shot          2022   
717                      Hold Your Breath: The Ice Dive          2022   
756                                      I Am Not Alone          2021   
767                      If Anything Happens I Love You          2020   
837                   John Was Trying to Contact Aliens          2020   
893                                       Larva Pendant          2022   
905                                        Lead Me Home          2021   
995                                   Making Unorthodox          2020   
1039                      Mighty Express: Train Trouble          2022   
1041              Mighty Little Bheem: I Love Taj Mahal          2022   
1101                             My Heroes Were Cowboys          2021   
1326                                        Robin Robin          2021   
1333                                  Rooting for Roona          2020   
1456                                        Sol Levante          2020   
1479                                 Spycraft: Season 1          2021   
1580                      The Boss Baby: Get That Baby!          2020   
1608                             The Claudia Kishi Club          2020   
1667    The Hand of God: Through the Eyes of Sorrentino          2021   
1729  The Magic School Bus Rides Again The Frizz Con...          2020   
1734                         The Martha Mitchell Effect          2022   
1807                                   The Speed Cubers          2020   
1809                                       The Stand-In          2020   
1916                                   Trese After Dark          2021   
2014                        What Would Sophia Loren Do?          2021   
2062                             You vs. Wild: Out Cold          2021   

      release_date  is_top10  \
46        20210914     False   
170       20210701     False   
218       20220127      True   
246       20220418     False   
274       20211124      True   
294       20211102      True   
302       20200211     False   
306       20200310      True   
311       20220222     False   
322       20210417     False   
529       20220217     False   
532       20211005     False   
647       20200501     False   
689       20210615     False   
691       20220217     False   
717       20220503     False   
756       20210916     False   
767       20201120     False   
837       20200820     False   
893       20220525      True   
905       20211130      True   
995       20200326     False   
1039      20220118     False   
1041      20220530      True   
1101      20210916     False   
1326      20211124     False   
1333      20201015      True   
1456      20200402     False   
1479      20210120      True   
1580      20200901      True   
1608      20200710     False   
1667      20211215     False   
1729      20201020     False   
1734      20220617     False   


In [98]:
len(movies_and_season_1_merge[movies_and_season_1_merge['_merge'] == 'both'].groupby('combined_title').filter(lambda x: len(x) > 1)['combined_title'].unique()) #125 titles have multiple matches oh no!

#multimatch = merge[merge['_merge'] == 'both'].groupby('combined_title').filter(lambda x: len(x) > 1)

125

In [75]:
manual_tv_matches = pd.read_csv("../data/interim/tv_matched.csv")
manual_tv_matches['title'] = clean_title(manual_tv_matches['title'])    

manual_movie_matches = pd.read_csv("../data/interim/movies_matched.csv")
manual_movie_matches['title'] = clean_title(manual_movie_matches['title'])

manual_other_matches = pd.read_csv('../data/raw/matched_wIMDB.tsv', sep='\t')
manual_other_matches['title'] = clean_title(manual_other_matches['title'])

In [77]:
multimatch['title'].drop_duplicates().isin(manual_tv_matches['title']).sum() #29
multimatch['title'].drop_duplicates().isin(manual_movie_matches['title']).sum() #28

28

In [85]:
netflix_data[netflix_data['title']=='after life']

combined_title  release_year  release_date  is_top10       title  \
67  After Life: Season 2          2020      20200424      True  after life   
68  After Life: Season 3          2022      20220114      True  after life   

    season  is_tv_show  
67     2.0        True  
68     3.0        True

In [87]:
imdb_all[imdb_all['tconst']=='tt8398600']

tconst titleType primaryTitle originalTitle  isAdult  release_year  \
651397  tt8398600  tvSeries   After Life    After Life        0        2019.0   

       endYear runtimeMinutes        genres       title  
651397    2022             30  Comedy,Drama  after life

In [66]:
multimatch.tail(20)

combined_title  release_year  release_date  is_top10  \
2493  We Can Be Heroes          2020      20201225      True   
2497    Wedding Season          2022      20220804      True   
2498    Wedding Season          2022      20220804      True   
2508           Whipped          2020      20200918     False   
2509           Whipped          2020      20200918     False   
2563   You've Got This          2020      20201002     False   
2564   You've Got This          2020      20201002     False   
2565     You: Season 3          2021      20211015      True   
2566     You: Season 3          2021      20211015      True   
2567     You: Season 3          2021      20211015      True   
2568     You: Season 3          2021      20211015      True   
2569     You: Season 3          2021      20211015      True   
2581    Zero: Season 1          2021      20210421     False   
2582    Zero: Season 1          2021      20210421     False   
2583    Zero: Season 1          2021      20210421     False   
2584    Zero: Season 1          2021      20210421     False   
2585    Zero: Season 1          2021      20210421     False   
2586    Zero: Season 1          2021      20210421     False   
2587    Zero: Season 1          2021      20210421     False   
2588    Zero: Season 1          2021      20210421     False   

                 title  season  is_tv_show      tconst titleType  \
2493  we can be heroes     NaN       False  tt11937426     short   
2497    wedding season     NaN       False  tt11426572     movie   
2498    wedding season     NaN       False  tt15114768  tvSeries   
2508           whipped     NaN       False  tt11754464     movie   
2509           whipped     NaN       False  tt12792836     short   
2563    youve got this     NaN       False  tt12635086     short   
2564    youve got this     NaN       False  tt13118012     movie   
2565               you     3.0        True  tt12497058     short   
2566               you     3.0        True  tt13849012     video   
2567               you     3.0        True  tt14530626     short   
2568               you     3.0        True  tt15169390     short   
2569               you     3.0        True  tt15484848     short   
2581              zero     1.0        True  tt10619486  tvSeries   
2582              zero     1.0        True  tt14012368     short   
2583              zero     1.0        True  tt14858664   tvMovie   
2584              zero     1.0        True  tt15072452     movie   
2585              zero     1.0        True  tt15137370     movie   
2586              zero     1.0        True  tt15398686     short   
2587              zero     1.0        True  tt15521636     video   
2588              zero     1.0        True  tt16205788     short   

          primaryTitle     originalTitle  isAdult endYear runtimeMinutes  \
2493  We Can Be Heroes  We Can Be Heroes      0.0      \N             \N   
2497    Wedding Season    Wedding Season      0.0      \N             98   
2498    Wedding Season    Wedding Season      0.0      \N             \N   
2508           Whipped             Bucin      0.0      \N             97   
2509           Whipped           Whipped      0.0      \N              7   
2563   You've Got This   You've Got This      0.0      \N             \N   
2564   You've Got This    Ahí te Encargo      0.0      \N            111   
2565               You               You      0.0      \N             \N   
2566               You               You      0.0      \N             \N   
2567              You!              You!      0.0      \N              7   
2568               YOU               You      0.0      \N             \N   
2569              YOU!              You!      0.0      \N              9   
2581              Zero              Zero      0.0    2021             27   
2582              Zero              Zero      0.0      \N             13   
2583              Zero              Zero      0.0      \N             90   
2584              

In [64]:
imdb_all

tconst titleType                 primaryTitle  \
0       tt0011801     movie             Tötet nicht mehr   
1       tt0013274     movie  Istoriya grazhdanskoy voyny   
2       tt0040241     short              Color Rhapsodie   
3       tt0044326     short                    Abstronic   
4       tt0044879     short                      Mandala   
...           ...       ...                          ...   
728908  tt9916724     short         Hay Que Ser Paciente   
728909  tt9916730     movie                       6 Gunn   
728910  tt9916756     short     Pretty Pretty Black Girl   
728911  tt9916764     short                           38   
728912  tt9916856     short                     The Wind   

                      originalTitle  isAdult  release_year endYear  \
0                  Tötet nicht mehr        0        2019.0      \N   
1       Istoriya grazhdanskoy voyny        0        2021.0      \N   
2                   Color Rhapsodie        0        2021.0      \N   
3                         Abstronic        0        2021.0      \N   
4                           Mandala        0        2021.0      \N   
...                             ...      ...           ...     ...   
728908         Hay Que Ser Paciente        0        2015.0      \N   
728909                       6 Gunn        0        2017.0      \N   
728910     Pretty Pretty Black Girl        0        2019.0      \N   
728911                           38        0        2018.0      \N   
728912                     The Wind        0        2015.0      \N   

       runtimeMinutes             genres                        title  
0                  \N       Action,Crime             tötet nicht mehr  
1                 133        Documentary  istoriya grazhdanskoy voyny  
2                   6              Short              color rhapsodie  
3                   6              Short                    abstronic  
4                   3              Short                      mandala  
...               ...                ...                          ...  
728908              3  Documentary,Short         hay que ser paciente  
728909            116                 \N                       6 gunn  
728910             \N              Short     pretty pretty black girl  
728911             \N              Short                           38  
728912             27              Short                     the wind  

[728913 rows x 10 columns]

In [5]:
##read imdb movie data
imdb_movie_data = pd.read_csv('../data/processed/imdb_movies.csv')

In [8]:
imdb_movie_data['title'] = clean_title(imdb_movie_data['title'])

In [9]:
#read matched data
matched_data = pd.read_csv('../data/raw/matched_wIMDB.tsv', sep='\t')[['tconst', 'title', 'release_year', 'season']]

In [13]:
imdb_clean_movie = imdb_movie_data[['tconst', 'title', 'startYear']].rename(columns={'startYear':'release_year'})

In [20]:
imdb_movie_data[imdb_movie_data['tconst']=='tt13353456']

Empty DataFrame
Columns: [tconst, titleType, primaryTitle, originalTitle, isAdult, startYear, endYear, runtimeMinutes, genres, title]
Index: []

In [18]:
imdb_clean_movie[imdb_clean_movie['tconst']=='tt13353456']

Empty DataFrame
Columns: [tconst, title, release_year]
Index: []

In [16]:
merge

combined_title  release_year  \
0                               #blackAF: Season 1          2020   
1                               (Un)Well: Season 1          2020   
2                         10 Days With Santa Claus          2020   
3                             100 Humans: Season 1          2020   
4                        1000 Miles from Christmas          2021   
...                                            ...           ...   
2256                    jeen-yuhs: A Kanye Trilogy          2022   
2257   the goop lab with Gwyneth Paltrow: Season 1          2020   
2258                            tick, tick...BOOM!          2021   
2259                                       Òlòtūré          2020   
2260  ​SAINT SEIYA: Knights of the Zodiac: Part II          2020   

      release_date  is_top10                                      title  \
0         20200417     False                                    blackaf   
1         20200812     False                                     unwell   
2         20201204     False                   10 days with santa claus   
3         20200313     False                                 100 humans   
4         20211224      True                  1000 miles from christmas   
...            ...       ...                                        ...   
2256      20220216      True                   jeenyuhs a kanye trilogy   
2257      20200124     False          the goop lab with gwyneth paltrow   
2258      20211119      True                              tick tickboom   
2259      20201002     False                                    òlòtūré   
2260      20200123      True  saint seiya knights of the zodiac part ii   

      season  is_tv_show      tconst     _merge  
0        1.0        True         NaN  left_only  
1        1.0        True         NaN  left_only  
2        NaN       False         NaN  left_only  
3        1.0        True         NaN  left_only  
4        NaN       False  tt13782052       both  
...      ...         ...         ...        ...  
2256     NaN       False         NaN  left_only  
2257     1.0        True         NaN  left_only  
2258     NaN       False         NaN  left_only  
2259     NaN       False         NaN  left_only  
2260     NaN       False         NaN  left_only  

[2261 rows x 9 columns]

In [7]:
imdb_clean2 = pd.concat([imdb_clean_movie, matched_data], axis=0)

In [8]:
movie_merge = netflix_data[~netflix_data['is_tv_show']].merge(imdb_clean2, on=['title', 'release_year'], how='left', indicator=True)

In [9]:
good_batch_1 = movie_merge[(movie_merge['_merge']=='both') & (movie_merge['tconst'].notnull())].groupby('title', as_index=False).filter(lambda x: len(x) == 1)[['tconst', 'title', 'release_year']]
good_batch_1.to_csv('../data/processed/good_batch_1.csv', index=False)
#824 unique matches on title alone
#Drops to 818 if match on year

In [11]:
manual_match_1 = movie_merge[(movie_merge['_merge']=='both') & (movie_merge['tconst'].notnull())].groupby('title', as_index=False).filter(lambda x: len(x) > 1)[['tconst', 'title', 'release_year']]
manual_match_1.to_csv('../data/processed/manual_match_1.csv', index=False)

In [12]:
manual_match_1.to_clipboard()

In [314]:
fails = movie_merge[movie_merge['_merge']=='left_only']

In [315]:
imdb_all = pd.read_csv('../data/processed/imdb_all.csv').rename(columns={'startYear':'release_year'})

In [316]:
movie_merge2 = fails[['title', 'release_year']].merge(imdb_all, on=['title', 'release_year'], how='left', indicator=True)

In [317]:
good_batch_2 = movie_merge2[(movie_merge2['_merge']=='both') & (movie_merge2['tconst'].notnull())].groupby('title', as_index=False).filter(lambda x: len(x) == 1)[['tconst', 'title', 'release_year']]
good_batch_2.to_csv('../data/processed/good_batch_2.csv', index=False)

In [318]:
manual_match_2 = movie_merge2[(movie_merge2['_merge']=='both') & (movie_merge2['tconst'].notnull())].groupby('title', as_index=False).filter(lambda x: len(x) > 1)[['tconst', 'title', 'release_year']].drop_duplicates()
manual_match_2.to_csv('../data/processed/manual_match_2.csv', index=False)

In [319]:
manual_match_3 = movie_merge2[movie_merge2['_merge']=='left_only'][['title', 'release_year']]
manual_match_3.to_csv('../data/processed/manual_match_3.csv', index=False)

In [102]:
print(f"Correctly Matched: {netflix_data[~netflix_data['is_tv_show']].shape[0] - good_batch_1.shape[0] - good_batch_2.shape[0]}\n")
print(f"Incorrectly Matched Duplicates Movies: {len(manual_match_1['title'].unique())}")
print(f"Incorrectly Matched Duplicate Non-movies: {len(manual_match_2['title'].unique())}")
print(f"Incorrectly Matched No Match: {len(manual_match_3['title'].unique())}")

#Remaining 119
#23 in manual_match_1
#6 in manual_match_2
#90 in manual_match_3


Correctly Matched: 119

Incorrectly Matched Duplicates Movies: 23
Incorrectly Matched Duplicate Non-movies: 6
Incorrectly Matched No Match: 90


## Repeat for TV Shows

In [192]:
#read imdb tv data
imdb_tv_data = pd.read_csv('../data/processed/imdb_tv.csv')[['tconst', 'title', 'startYear']].rename(columns={'startYear':'release_year'})

1021 Shows

In [210]:
#netflix_data[netflix_data['is_tv_show']].shape

(1121, 7)

In [234]:
tv_merge1 = netflix_data[netflix_data['is_tv_show']].merge(imdb_tv_data, on=['title'], how='left', indicator=True)

Somehow we already lost 4. I know 3 are because of duplicate title + season, but not sure where the last 1 went.

Left with 1117

In [235]:
#tv_merge1[['title', 'season']].drop_duplicates().shape

In [236]:
good_batch_tv_1 = tv_merge1[(tv_merge1['_merge']=='both') & (tv_merge1['tconst'].notnull())].groupby(['title', 'season'], as_index=False).filter(lambda x: len(x) == 1)[['tconst', 'title', 'season', 'release_year_x']].rename(columns={'release_year_x':'release_year'})
manual_match_tv_1 = tv_merge1[(tv_merge1['_merge']=='both') & (tv_merge1['tconst'].notnull())].groupby(['title', 'season'], as_index=False).filter(lambda x: len(x) > 1)[['tconst', 'title', 'season', 'release_year_x']].rename(columns={'release_year_x':'release_year'})

In [245]:
print(f"Matched 1:1: {good_batch_tv_1.shape[0]}")
print(f"Matched 1:M: {manual_match_tv_1[['title', 'season']].drop_duplicates().shape[0]}")

Matched 1:1: 878
Matched 1:M: 154


Immediately 878 match 1:1, 154 match 1:M, 85 don't match. Everything OK up to here

In [247]:
#chip away at some of the manual_tv_1 matches using zach match
#already_matched_titles = manual_match_tv_1[manual_match_tv_1['title'].isin(matched_data['title'])]['title'].unique()
#manual_match_tv_2 = manual_match_tv_1[~manual_match_tv_1['title'].isin(already_matched_titles)]
#manual_match_tv_2.to_csv('../data/processed/manual_match_tv_1.csv', index=False)

In [218]:
#add the already matched to the good batch
#already_matched_df = matched_data[matched_data['title'].isin(already_matched_titles)][['tconst', 'title', 'season', 'release_year']]
#good_batch_tv_1 = pd.concat([good_batch_tv_1, already_matched_df], axis=0)
#good_batch_tv_1.to_csv('../data/processed/good_batch_tv_1.csv', index=False)

Of the 154 1:M, 9 are already matched via Zach's list. 145 remain

In [253]:
#9 matches
good_batch_tv_2 = manual_match_tv_1[['title', 'season', 'release_year']].merge(matched_data, on=['title', 'season', 'release_year'], how='inner').drop_duplicates()
good_batch_tv_2.to_csv('../data/processed/good_batch_tv_2.csv', index=False)

Of the 145 remaining, 83 match on season 1 release year. 62 remaining

In [291]:
#Clean more manual matches using release year with season 1
good_batch_tv_3 = manual_match_tv_1[['title', 'season', 'release_year']].drop_duplicates().merge(imdb_tv_data, on=['title', 'release_year'], how='inner')
good_batch_tv_3 = good_batch_tv_3.groupby(['title', 'season'], as_index=False).filter(lambda x: len(x) == 1)[['tconst', 'title', 'season', 'release_year']]
good_batch_tv_3.to_csv('../data/processed/good_batch_tv_3.csv', index=False)
print(good_batch_tv_3.shape)

(83, 4)


62 remaining manual matches with potential ttvalues

In [298]:
checker = pd.concat([good_batch_tv_2, good_batch_tv_3], axis=0).merge(manual_match_tv_1, on=['title', 'season'], how='outer', indicator=True)
remaining_manual_1 = checker[checker['_merge']=='right_only'][['title', 'season', 'release_year_y', 'tconst_y']].rename(columns={'release_year_y':'release_year', 'tconst_y':'tconst'})

remaining_manual_1.to_csv('../data/processed/manual_match_tv_1_updated.csv', index=False)

In [221]:
# 1121 tv shows
# 887 matched 
# 145 multiple ttcodes


Move on to the 85 non-matches

In [246]:
fails_tv = tv_merge1[tv_merge1['_merge']=='left_only']
print(fails_tv.shape)

(85, 10)


In [300]:
#Now try matching failed matches using zach's data
tv_merge2 = fails_tv[['title', 'season', 'release_year_x']].merge(matched_data, on=['title', 'season'], how='left', indicator=True)
good_batch_tv_4 = tv_merge2[(tv_merge2['_merge']=='both') & (tv_merge2['tconst'].notnull())].groupby(['title', 'season'], as_index=False).filter(lambda x: len(x) == 1)[['tconst', 'title', 'season', 'release_year']].rename(columns={'release_year_x':'release_year'})
manual_match_tv_2 = tv_merge2[(tv_merge2['_merge']=='left_only')][['title', 'season', 'release_year_x']].rename(columns={'release_year_x':'release_year'})

66 Match with Zach's list. 19 remaining

In [302]:
good_batch_tv_4.to_csv('../data/processed/good_batch_tv_2.csv', index=False)
#66 good matches

19 Remaining need a manual match

In [304]:
#manual_match_tv_2.shape
manual_match_tv_2.to_csv('../data/processed/manual_match_tv_2.csv', index=False)

Combined Required Matches into 1

In [305]:
pd.concat([remaining_manual_1, manual_match_tv_2], axis=0).to_csv('../data/interim/tv_need_matches.csv', index=False)

In [320]:
pd.concat([manual_match_1, manual_match_2, manual_match_3], axis=0).to_csv('../data/interim/movies_need_matches.csv', index=False)

In [ ]:
pd.read_csv('../data/interim/movies_need_matches.csv').shape